scio-core/src/main/scala/com/spotify/scio/coders/Coder.scala

```diff
@@ -170,21 +170,45 @@ private final case class DisjunctionCoder[T, Id](
     }
 }
 
-private[scio] final class RefCoder[T](val typeName: String, c: => BCoder[T]) extends BCoder[T] {
+private[scio] final class RefCoder[T](val typeName: String, var c: BCoder[T]) extends BCoder[T] {
   def value: BCoder[T] = c
```

Comment:
since it's a `var` now we could remove this and probably have a better name for `c`?